doc/src/sphinx/develop/extractors.rst

```diff
@@ -27,4 +34,268 @@ The current list of supported events is:
+
+.. code-block:: bash
+
+	docker run -ti --rm --network tests_clowder clowder/mongo-init
+
+
+4. Enter email, first name, last name password, and admin: true when prompted.
+
+5. Navigate to localhost:9000 and login with credentials you created in step 4.
+
+6. Create a test space and dataset. Then click 'Select Files' and upload (if the file stays in CREATED and never moves to PROCESSED you might need to change the permission on the data folder using docker run -ti --rm --network tests_clowder clowder/mongo-init).
+
+7. Click on file and type submit for extraction.
+
+8. It may take a few minutes for you to be able to see the extractors available within Clowder.
+
+9. Eventually you should see your extractor in the list and click submit.
+
+10. Navigate back to file and click on metadata.
+
+11. You should see your metadata present if all worked successfully.
+
+
+A Quick Note on Debugging
+##########################
+
+To check the status of your extraction, navigate to the file within Clowder and click on the “Extractions” tab. This will give you a list of extractions that have been submitted. Any error messages will show up here if your extractor did not run successfully. 
+
+.. container:: imagepadding
+    .. image:: /_static/ug_extractors-1.png
+
+You can expand the tab to see all submissions of the extractor and any error messages associated with the submission:
+
+.. container:: imagepadding
+    .. image:: /_static/ug_extractors-1.png
+
+If your extractor failed, the error message is not helpful, or if you do not see metadata present in the “Metadata” tab for the file you can check the logs of your extractor coming from the docker container by executing the following:
+
+.. code-block:: bash
+
+	docker log tests_myextractor_1 
+
+
+Replace “myextractor” with whatever name you gave your extractor in the docker-compose.extractors.yml file.
+
+If you want to watch the logs as your extractor is running you can type:
+
+.. code-block:: bash
+
+	docker logs -f tests_myextractor_1
+
+.. container:: imagepadding
+    .. image:: /_static/ug_extractors-1.png
+
+You can print any debugging information within your extractor to the docker logs by utilizing the logging object within your code. The following example is for pyClowder:
+
+.. code-block:: bash
+
+	logging.info("Uploaded metadata %s", metadata)
+
+
+In the screenshot above you can see the lines printed out by the logging.info as the line will start with INFO:
+
+.. code-block:: bash
+
+	2021-04-27 16:47:49,995 [MainThread     ] INFO
+
+
+Additional pyClowder Examples
+##############################
+
+For a simple example of an extractor, please refer to `extractor-csv <https://opensource.ncsa.illinois.edu/bitbucket/projects/CATS/repos/extractors-csv/browse>`_. This extractor is submitted on a csv file and returns the headers as metadata.
```

Comment:
use https://github.com/clowder-framework/extractors-csv

Reply:
I've committed a change for this